tests/jobs/scala/src/main/scala/ShuffleApp.scala

```diff
@@ -0,0 +1,49 @@
+import org.apache.spark.sql.SparkSession
+
+import scala.util.Random
+
+/**
+  * This is a slight modification of GroupByTest from Spark Examples but with a deterministic
+  * sequential key generation which doesn't rely on random number generator.
+  *
+  * Usage: ShuffleApp [numMappers] [numKeys] [valueSize] [numReducers] [sleepBeforeShutdown]
+  */
+object ShuffleApp {
+  def main(args: Array[String]) {
+    val spark = SparkSession
+      .builder
+      .appName("Shuffle Test")
+      .getOrCreate()
+
+    val numMappers = if (args.length > 0) args(0).toInt else 2
+    val numKeys = if (args.length > 1) args(1).toInt else 1000
+    val valSize = if (args.length > 2) args(2).toInt else 1000
+    val numReducers = if (args.length > 3) args(3).toInt else numMappers
+    val sleepBeforeShutdown = if (args.length > 4) args(4).toInt else 0
+
+    val pairs1 = spark.sparkContext.parallelize(0 until numMappers, numMappers).flatMap { _ =>
+      val ranGen = new Random
+      val arr1 = new Array[(Int, Array[Byte])](numKeys)
```

Comment:
nit: can you drop the `1` from `arr1` or rename to something a little more descriptive?